skyfield/jpllib.py

```diff
@@ -204,8 +206,21 @@ def __repr__(self):
 
 class ChebyshevPosition(SPICESegment):
     def _at(self, t):
-        position, velocity = self.spk_segment.compute_and_differentiate(t.tdb)
-        return position / AU_KM, velocity / AU_KM, None, None
+        try:
+            position, velocity = self.spk_segment.compute_and_differentiate(t.tdb)
+            return position / AU_KM, velocity / AU_KM, None, None
+        except OutOfRangeError as e:
+            start_time = t.ts.tdb(jd=self.spk_segment.start_jd)
+            end_time = t.ts.tdb(jd=self.spk_segment.end_jd)
+            out_of_range_times = []
+
+            for i, time in enumerate(t):
```

Comment:
I agree that we want to give users information about which times were the problem, but I'm not sure an array of separate `Time` objects, one per value, will be very efficient — if they passed us an array of several thousand times, most of which were out of range, this loop would take a bit of time to create thousands of `Time` objects.

It would be simplest to offer users the bare Boolean mask `e.out_of_range_times` (it's a list of Trues and Falses, if I'm reading right?) as an attribute. Then they would be free to process that information as they wish, and if they didn't need separate `Time` objects created for every bad value, they'd be saved the expense.

Plus: they would know immediately which array elements were bad. If we simply return the values of the bad times, then they'd have to search their time array manually for each value to know where they are.